zello-sdk-sample-misc/src/main/java/com/zello/sdk/sample/misc/MiscActivity.java

```diff
@@ -218,147 +219,147 @@ public void onMicrophonePermissionNotGranted() {
+		builder.setSingleChoiceItems(items, selection, new DialogInterface.OnClickListener() {
+			@Override
+			public void onClick(DialogInterface dialog, int which) {
+				switch (which) {
+					case 0: {
+						Zello.getInstance().setStatus(com.zello.sdk.Status.AVAILABLE);
+						break;
+					}
+					case 1: {
+						Zello.getInstance().setStatus(com.zello.sdk.Status.SOLO);
+						break;
+					}
+					case 2: {
+						Zello.getInstance().setStatus(com.zello.sdk.Status.BUSY);
+						break;
+					}
+					case 3: {
+						createCustomStatusMessageDialog();
+						break;
+					}
+				}
+				dialog.dismiss();
+			}
+		});
+		builder.setCancelable(true).setTitle(res.getString(R.string.menu_set_status));
+		final AlertDialog dialog = builder.create();
+		dialog.setCanceledOnTouchOutside(true);
+		dialog.show();
+	}
+
+	private void updateUIForLocked() {
+		_lockButton.setText(R.string.unlock);
+	}
+
+	private void updateUIForUnlocked() {
+		_lockButton.setText(R.string.lock);
+	}
+
+	private void updateUIForSignedIn() {
+		invalidateOptionsMenu();
+
+		_externalIdEditText.setVisibility(View.VISIBLE);
+		_externalIdEditText.setText(_appState.getExternalId());
+
+		_externalIdTextView.setVisibility(View.VISIBLE);
+
+		_statusTextView.setVisibility(View.INVISIBLE);
+
+		if (_appState.isLocked()) {
+			updateUIForLocked();
+		} else {
+			updateUIForUnlocked();
+		}
+
+		_lockButton.setVisibility(View.VISIBLE);
+	}
+
+	private void updateUIForSigningIn() {
+		invalidateOptionsMenu();
+
+		_statusTextView.setVisibility(View.VISIBLE);
+		_lockButton.setVisibility(View.INVISIBLE);
+		_externalIdEditText.setVisibility(View.INVISIBLE);
+		_externalIdTextView.setVisibility(View.INVISIBLE);
+
+		_statusTextView.setText(R.string.sign_in_status_signing_in);
+	}
+
+	private void updateUIForDisconnected() {
+		invalidateOptionsMenu();
+
+		_statusTextView.setVisibility(View.VISIBLE);
+		_lockButton.setVisibility(View.INVISIBLE);
+		_externalIdEditText.setVisibility(View.INVISIBLE);
+		_externalIdTextView.setVisibility(View.INVISIBLE);
+
+		_statusTextView.setText(R.string.sign_in_status_offline);
+	}
+
+	private void createCustomStatusMessageDialog() {
+		AlertDialog.Builder builder = new AlertDialog.Builder(this);
+		Zello.getInstance().getAppState(_appState);
+		String previousStatus = _appState.getStatusMessage();
+		final EditText statusEditText = new EditText(this);
+		statusEditText.setText(previousStatus);
+		builder.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+			@Override
+			public void onClick(DialogInterface dialog, int which) {
+				Zello.getInstance().setStatusMessage(statusEditText.getText().toString());
+			}
+		});
+		builder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+			@Override
+			public void onClick(DialogInterface dialog, int which) {
+
```

Comment:
Is an empty impl correct? Seems like it should dismiss the dialog

Reply:
Not necessary